### PR TITLE
Manager

### PR DESCRIPTION
**Pull request by PotmanNob**
w manager trust

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Manager/Manager.java [hunk 1]
```diff
@@ -39,15 +40,35 @@ private Manager() {
             addRunnableTrigger(() -> {commandScheduler.schedule(sysIdSubsystem.sysIdQuasistatic(Direction.kForward));}, () -> TEST_CONTROLLER.getXButtonPressed());
             addRunnableTrigger(() -> {commandScheduler.schedule(sysIdSubsystem.sysIdQuasistatic(Direction.kReverse));}, () -> TEST_CONTROLLER.getYButtonPressed());
         }
-        addTrigger(ManagerStates.IDLE, ManagerStates.SCORING_HIGH, () -> Controllers.DRIVER_CONTROLLER.getXButtonPressed());
-        addTrigger(ManagerStates.SCORING_HIGH, ManagerStates.IDLE, () -> Controllers.DRIVER_CONTROLLER.getXButtonPressed());
+        //From idle
```
Comment (nkalupahana) [inline src/main/java/frc/robot/subsystems/Manager/Manager.java:43]: nit: comments should have spaces after them, e.g.
```suggestion
        // From idle
```

@GreenTomato5 do y'all not have an autoformatter?

File: src/main/java/frc/robot/subsystems/Manager/Manager.java [hunk 2]
```diff
@@ -58,13 +79,19 @@ public void runState() {
...
         intake.setState(getState().getIntakeState());
+        indexer.setState(getState().getIndexerState());
 
         drive.periodic();
         vision.periodic();
         elevator.periodic();
         intake.periodic();
+        indexer.periodic();
 
         // Other subsystem periodics go here
```
Comment (nkalupahana) [inline src/main/java/frc/robot/subsystems/Manager/Manager.java:96]:
```suggestion
```

**Comment by nkalupahana** [comment]
No idea how much free time y'all have, but it would be so cool to be able to take a subsystem's trigger setup and export a state machine diagram for it to make it easier to visualize. You could have it output some sort of JSON object on init with the different triggers (to SmartDashboard or to the console or to a file or whatever), and then pass that to a pretty simple Python script? ChatGPT could def do most of the work. Would highly recommend, bc this is pretty complex and difficult to parse just by reading the code, and it's also hard to make sure you didn't miss anything

File: src/main/java/frc/robot/subsystems/Manager/Manager.java [hunk 3]
```diff
@@ -58,13 +79,19 @@ public void runState() {
         Logger.recordOutput("Manager/State", getState().getStateString());
         Logger.recordOutput("Manager/State Time", getStateTime());
 
+        if(Controllers.OPERATOR_CONTROLLER.getXButtonPressed()) {
```
Comment (nkalupahana) [inline src/main/java/frc/robot/subsystems/Manager/Manager.java:82]: This should be a runnable trigger, right? Ideally no `if` statements in the `runState` function

File: src/main/java/frc/robot/subsystems/Indexer/IndexerIOSim.java [hunk 4]
```diff
@@ -18,7 +18,7 @@ public class IndexerIOSim implements IndexerIO {
     private boolean[] beamBreakArray = {false, false, false, false, false, false};
 
     public IndexerIOSim() {
-        wheelMotorSim = new DCMotorSim(LinearSystemId.createDCMotorSystem(DCMotor.getKrakenX60(1), 0.000001, 1), DCMotor.getKrakenX60Foc(IndexerConstants.NUM_SPINNER_MOTORS), 0);
+        wheelMotorSim = new DCMotorSim(LinearSystemId.createDCMotorSystem(DCMotor.getKrakenX60(1), 0.000001, 1), DCMotor.getKrakenX60Foc(IndexerConstants.NUM_SPINNER_MOTORS));
```
Comment (GreenTomato5) [inline src/main/java/frc/robot/subsystems/Indexer/IndexerIOSim.java:21]: if ur using krakenFOC you might have to change the motor request type to FOC for the talonfxs in the real IO

File: src/main/java/frc/robot/subsystems/Manager/Manager.java [hunk 5]
```diff
@@ -58,13 +79,19 @@ public void runState() {
         Logger.recordOutput("Manager/State", getState().getStateString());
         Logger.recordOutput("Manager/State Time", getStateTime());
 
+        if (Controllers.OPERATOR_CONTROLLER.getXButtonPressed()) {
```
Comment (GreenTomato5) [inline src/main/java/frc/robot/subsystems/Manager/Manager.java:82]: this should be a runnable trigger

File: src/main/java/frc/robot/subsystems/Manager/Manager.java [hunk 6]
```diff
@@ -58,13 +79,19 @@ public void runState() {
         Logger.recordOutput("Manager/State", getState().getStateString());
```
Comment (GreenTomato5) [inline src/main/java/frc/robot/subsystems/Manager/Manager.java:79]: subsystem name in the subsystem constants

File: src/main/java/frc/robot/subsystems/Manager/Manager.java [hunk 7]
```diff
@@ -39,15 +40,35 @@ private Manager() {
             addRunnableTrigger(() -> {commandScheduler.schedule(sysIdSubsystem.sysIdQuasistatic(Direction.kForward));}, () -> TEST_CONTROLLER.getXButtonPressed());
             addRunnableTrigger(() -> {commandScheduler.schedule(sysIdSubsystem.sysIdQuasistatic(Direction.kReverse));}, () -> TEST_CONTROLLER.getYButtonPressed());
         }
-        addTrigger(ManagerStates.IDLE, ManagerStates.SCORING_HIGH, () -> Controllers.DRIVER_CONTROLLER.getXButtonPressed());
-        addTrigger(ManagerStates.SCORING_HIGH, ManagerStates.IDLE, () -> Controllers.DRIVER_CONTROLLER.getXButtonPressed());
+        // From idle
+        addTrigger(ManagerStates.IDLE, ManagerStates.INTAKING, () -> Controllers.DRIVER_CONTROLLER.getBButtonPressed());
```
Comment (GreenTomato5) [inline src/main/java/frc/robot/subsystems/Manager/Manager.java:44]: have you tested all the scoring modes in sim?

**Review by GreenTomato5 — CHANGES_REQUESTED** [review]
U wana do that diagraming thing? or someone else @PotmanNob

File: src/main/java/frc/robot/subsystems/Manager/Manager.java [hunk 8]
```diff
@@ -39,15 +40,35 @@ private Manager() {
...
+        addTrigger(ManagerStates.IDLE, ManagerStates.OUTTAKING, () -> Controllers.DRIVER_CONTROLLER.getAButtonPressed());
+        addTrigger(ManagerStates.IDLE, ManagerStates.GOING_HIGH, () -> Controllers.DRIVER_CONTROLLER.getRightTriggerAxis() > Controllers.TRIGGERS_REGISTER_POINT);
+        addTrigger(ManagerStates.IDLE, ManagerStates.GOING_MID, () -> Controllers.DRIVER_CONTROLLER.getLeftTriggerAxis() > Controllers.TRIGGERS_REGISTER_POINT);
```
Comment (GreenTomato5) [inline src/main/java/frc/robot/subsystems/Manager/Manager.java:47]: This seems pretty hard to control, why not just use d pad mid and high to score?

File: src/main/java/frc/robot/subsystems/Manager/Manager.java [hunk 9]
```diff
@@ -39,33 +40,57 @@ private Manager() {
...
             instance = new Manager();
         }
         return instance;
     }
 
     @Override
     public void runState() {
-        Logger.recordOutput("Manager/State", getState().getStateString());
-        Logger.recordOutput("Manager/State Time", getStateTime());
+        Logger.recordOutput(ManagerConstants.SUBSYSTEM_NAME + "/State", getState().getStateString());
```
Comment (GreenTomato5) [inline src/main/java/frc/robot/subsystems/Manager/Manager.java:82]: you coulldddd static import (Not needed)

**Review by GreenTomato5 — APPROVED** [review]
Yeah this looks chill if it works in sim

File: src/main/java/frc/robot/subsystems/Manager/Manager.java (hunk 8 above)
Comment (PotmanNob) [reply src/main/java/frc/robot/subsystems/Manager/Manager.java]: Dpad works weirdly in wpilib and i think it will make the controls harder to use

**Review by GreenTomato5 — APPROVED** [review]
🥇 💯 🎂

**Pull request merged**